### PR TITLE
Add paddle.geometric

### PR DESCRIPTION
**Pull request by DesmonDay**
### PR types
Performance optimization

### PR changes
Others

### Describe
Add paddle.geometric in PGL.

### CODE REVIEW

File: pgl/nn/conv.py [hunk 1]
```diff
@@ -349,14 +332,26 @@ def forward(self, graph, feature):
...
-        msg = graph.send(
-            self._send_attention,
-            src_feat={"src": attn_src,
-                      "h": feature},
-            dst_feat={"dst": attn_dst}, )
-        output = graph.recv(reduce_func=self._reduce_attention, msg=msg)
+        alpha = paddle.geometric.send_uv(attn_src, attn_dst, src_index,
+                                         dst_index, "add")
+        alpha = self.leaky_relu(alpha)
+        alpha = math.segment_softmax(alpha, segment_ids)
```
Comment (DesmonDay) [inline pgl/nn/conv.py:342]: graph.apply_edge

Comment (DesmonDay) [reply pgl/nn/conv.py:342]: edge_softmax(graph, xxx)

File: pgl/nn/conv.py [hunk 2]
```diff
@@ -349,14 +332,26 @@ def forward(self, graph, feature):
...
+        alpha = paddle.reshape(alpha, [-1, self.num_heads, 1])
+        if self.attn_drop > 1e-15:
+            alpha = self.attn_dropout(alpha)
+
+        output = paddle.geometric.send_ue_recv(feature, alpha, src_index,
```
Comment (DesmonDay) [inline pgl/nn/conv.py:347]: 写到graph里

Comment (DesmonDay) [inline pgl/nn/conv.py:347]: DistGPUGraph

Comment (DesmonDay) [reply pgl/nn/conv.py:347]: DistGPUGraph

File: pgl/utils/edge_index.py [hunk 3]
```diff
@@ -134,36 +136,56 @@ def is_tensor(self):
...
+        if not paddle.device.is_compiled_with_cuda() and uva:
+            raise ValueError("uva mode should be run under gpu environment!")
+
         if inplace:
-            self._sorted_u = paddle.to_tensor(self._sorted_u)
-            self._sorted_v = paddle.to_tensor(self._sorted_v)
-            self._sorted_eid = paddle.to_tensor(self._sorted_eid)
-            self._degree = paddle.to_tensor(self._degree)
-            self._indptr = paddle.to_tensor(self._indptr)
+            if not uva:
```
Comment (Yelrose) [inline pgl/utils/edge_index.py:162]: 建议写个utils.py to_paddle_tensor 有个flag解决uva的问题

**Review by Yelrose — APPROVED** [review]
LGTM

**Pull request merged**